pkg/render/prometheus_service.go

```diff
@@ -0,0 +1,253 @@
+	path := p.installation.ImagePath
+	prefix := p.installation.ImagePrefix
+
+	prometheusServiceImage, err := components.GetReference(components.ComponentTigeraPrometheusService, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	p.prometheusServiceImage = prometheusServiceImage
+
+	return nil
+}
+
+// Objects returns the lists of objects in this component that should be created and/or deleted during
+// rendering.
+func (p *tigeraPrometheusServiceComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
+	// tigera-prometheus-objects
+	namespacedObjects := []client.Object{}
+
+	// place pullsecrets secrets under tigera-prometheus
+	secrets := secret.CopyToNamespace(common.TigeraPrometheusNamespace, p.pullSecrets...)
+	namespacedObjects = append(namespacedObjects, secret.ToRuntimeObjects(secrets...)...)
+
+	namespacedObjects = append(
+		namespacedObjects,
+		p.tigeraPrometheusServiceDeployment(),
+		p.calicoNodePrometheusService(),
+	)
+
+	objsToCreate = []client.Object{}
+	objsToCreate = append(objsToCreate, namespacedObjects...)
+
+	objsToDelete = []client.Object{}
+
+	return objsToCreate, objsToDelete
+
+}
+
+func (p *tigeraPrometheusServiceComponent) Ready() bool {
+	return true
+}
+
+func (p *tigeraPrometheusServiceComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+// calicoNodePrometheusService sets up a service for the Prometheus Service/Proxy deployment
+func (p *tigeraPrometheusServiceComponent) calicoNodePrometheusService() *corev1.Service {
+
+	s := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      calicoNodePrometheusServiceName,
+			Namespace: common.TigeraPrometheusNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"k8s-app": tigeraPrometheusServiceName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "web",
+					Port:       PrometheusDefaultPort,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(p.prometheusServicePort),
+				},
+			},
+		},
+	}
+
+	return s
+}
+
+// tigeraPrometheusServiceDeployment deployment for the Prometheus Service/Proxy pod and image
+func (p *tigeraPrometheusServiceComponent) tigeraPrometheusServiceDeployment() *appsv1.Deployment {
+	var replicas int32 = 1
+	podDnsPolicy := corev1.DNSClusterFirst
+	podHostNetworked := false
+
+	// set hostnetworked if EKS with Calico CNI due to EKS to allow other Pods to
+	// connecto to tigera-prometheus-service since Calico does not manage node network
+	// in EKS - uses Amazon VPC
+	if p.installation.KubernetesProvider == operatorv1.ProviderEKS &&
+		p.installation.CNI.Type == operatorv1.PluginCalico {
+		podHostNetworked = true
+		// corresponding dns policy for hostnetwoked pods to resolve the service
+		// hostname urls
+		podDnsPolicy = corev1.DNSClusterFirstWithHostNet
+	}
+
+	d := &appsv1.Deployment{
```

Comment:
@mgleung do you think we need some PSP's for this?

Reply:
For what it is worth, PSPs are deprecated and are being removed. https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/

The replacement isn't available yet, so maybe still need them here.